src/NSwag.CodeGeneration.CSharp/Models/CSharpClientTemplateModel.cs

```diff
@@ -129,7 +129,14 @@ public string JsonSerializerParameterCode
                 if (RequiresJsonExceptionConverter)
                     jsonConverters = jsonConverters.Concat(new[] { "JsonExceptionConverter" });
 
-                return CSharpJsonSerializerGenerator.GenerateJsonSerializerParameterCode(handleReferences, jsonConverters.ToList());
+                var parameterCode = CSharpJsonSerializerGenerator.GenerateJsonSerializerParameterCode(handleReferences, jsonConverters.ToList());
+
+                if (parameterCode.Length == 0)
+                    parameterCode = "new Newtonsoft.Json.JsonSerializerSettings()";
+                else
+                    parameterCode = parameterCode.Substring(2);
```

Comment:
Why do you transform the code? This way the handleReferences setting is not respected... We should directly return `parameterCode.Substring(2); ` (and I should fix GenerateJsonSerializerParameterCode [remove ", "]).